spec/project_printer_spec.rb

```diff
@@ -0,0 +1,70 @@
+require "spec_helper"
+require "project_printer"
+
+RSpec.describe ProjectPrinter do
+  describe "#changed_name" do
+    it "changes the project name" do
+      stdout = StringIO.new("")
+      project_name = "First project"
+      new_project_name = "Updated project"
+      project_printer = ProjectPrinter.new(stdout)
+      output = "\e[38;5;40mChanged project name from\e[0m 'First project' \e[38;5;40mto\e[0m 'Updated project'"
```

Comment:
This is an ugly bit of string assertion! Is the formatting important to the assertion? If not then you can use a regex for the important part or strip out the output before making an assertion so it is just the content.

If the formatting is important, then identifying what is important about those formatting strings and naming them would be a way to test this. For example, you could name this "\e[38;5;40m" something that characterized what it was conveying in a variable. Then you can make assertions with the variable.